### PR TITLE
Add a get method to the actor interface and add snapshot app which queries it

### PR DESCRIPTION
**Pull request by LMWF**
# Description

Add a get method to the actor interface and add snapshot app which queries it.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #15

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: snapshot/snapshot.sln [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (amanbha) [inline snapshot/snapshot.sln:2]: Overall LGTM,  rather than adding individual solutions, please add a solution at root with all projects in them, it would be easier to manage.

File: snapshot/Dockerfile.original [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# ------------------------------------------------------------
```
Comment (amanbha) [inline snapshot/Dockerfile.original:1]: is this file needed?

File: snapshot/Models/HTTPResponse.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Tests.HashTagApp.Models
+{
+    using System.ComponentModel.DataAnnotations;
+
+    public class HTTPResponse
```
Comment (amanbha) [inline snapshot/Models/HTTPResponse.cs:10]: Where is this class used?

File: snapshot/Models/SocialMediaMessage.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Tests.HashTagApp.Models
+{
+    using System;
+    using System.ComponentModel.DataAnnotations;
+
+    public class SocialMediaMessage
```
Comment (amanbha) [inline snapshot/Models/SocialMediaMessage.cs:11]: where is this class used?

File: snapshot/Program.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    using Microsoft.Extensions.Logging;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+
+    public class Program
+    {
+        // This uses the names of shapes for a generic theme
+        static internal string[] HashTags = new string[]
```
Comment (amanbha) [inline snapshot/Program.cs:24]: this is also in https://github.com/dapr/test-infra/blob/master/feed-generator/Program.cs
can be moved to a common constants or file can be source included.

File: .github/workflows/snapshot.yml [hunk 6]
```diff
@@ -0,0 +1,78 @@
+# ------------------------------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
```
Comment (amanbha) [inline .github/workflows/snapshot.yml:3]: At this point we have added too many yaml files here which are very similar, its a good time to merge these.

Comment (artursouza) [reply .github/workflows/snapshot.yml:3]: Second that.

File: snapshot/Program.cs [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+            "limocon",
+            "hypocycoid"
+        };
+
+        private static string[] Sentiments = new string[]
+        {
+            "verynegative",
+            "negative",
+            "neutral",
+            "strong",
```
Comment (artursouza) [inline snapshot/Program.cs:43]: positive and verypositive instead of "strong".

File: snapshot/snapshot.sln (hunk 1 above)
Comment (artursouza) [reply snapshot/snapshot.sln:2]: I agree.

File: snapshot/Program.cs [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+            "verystrong"
+        };
+
+        public static void Main(string[] args)
```
Comment (artursouza) [inline snapshot/Program.cs:47]: This is not really a test. It is just a way to run these locally, right? Do we have unit tests?

File: snapshot/snapshot.sln (hunk 1 above)
Comment (youngbupark) [reply snapshot/snapshot.sln:2]: +1, So we have the common model class which is called `SocialMediaMessage`. I would create classlib project to hold this common models instead of defining multiple times in the multiple apps.

File: snapshot/Program.cs (hunk 8 above)
Comment (LMWF) [reply snapshot/Program.cs:47]: I'm not sure what this comment is pointing at, it looks like it's pointing at Main()?

It's not a test in itself, it's an app that's part of the pipeline.  There's no unit tests, every app really just does one thing.

File: snapshot/Dockerfile.original (hunk 2 above)
Comment (LMWF) [reply snapshot/Dockerfile.original:1]: Deleted.

File: snapshot/Models/HTTPResponse.cs (hunk 3 above)
Comment (LMWF) [reply snapshot/Models/HTTPResponse.cs:10]: Removed.

File: snapshot/Models/SocialMediaMessage.cs (hunk 4 above)
Comment (LMWF) [reply snapshot/Models/SocialMediaMessage.cs:11]: Removed

File: snapshot/Program.cs (hunk 5 above)
Comment (LMWF) [reply snapshot/Program.cs:24]: Yes I agree, but it's extra work to get the images building if there's a common folder outside.  I'd like to do that later.

File: .github/workflows/snapshot.yml (hunk 6 above)
Comment (LMWF) [reply .github/workflows/snapshot.yml:3]: How would you merge these?  I don't think these can be templatized w/o helm.

File: snapshot/Program.cs (hunk 5 above)
Comment (amanbha) [reply snapshot/Program.cs:24]: All the common files can be moved to a common .dll or can be source included from a common folder

File: .github/workflows/snapshot.yml (hunk 6 above)
Comment (youngbupark) [reply .github/workflows/snapshot.yml:3]: @LMWF we can use the single workflow. but you need to add the step to parse changed app. I can sync up with you.

**Comment by youngbupark** [comment]
One suggestion to remove the duplicated codes:
1. Dockerfiles - use build-arguments
2. GH Actions workflows - we can add additional step to parse changed directory from commit.
3. common model project - I see the same SocialMediaMessage model definitions here and there. we can create the common classlib project and make all apps referring to it.

File: snapshot/Program.cs (hunk 5 above)
Comment (LMWF) [reply snapshot/Program.cs:24]: Opened https://github.com/dapr/test-infra/issues/47

File: .github/workflows/snapshot.yml (hunk 6 above)
Comment (LMWF) [reply .github/workflows/snapshot.yml:3]: Opened

Comment (amanbha) [reply .github/workflows/snapshot.yml:3]: Sure, lets use the new issue to track these improvements, in the interest of getting the first run up and running, I am fine checking it in and track the improvement in the new issue, but lets try to tackle them in this milestone.
